playbooks/vars/oci/profiles/ha/network.yaml

```diff
@@ -0,0 +1,332 @@
+
+    security_groups: "{{ security_groups | d([]) }}"
+    tags: "{{ cluster_state.tags | d({}) }}"
+
+    spec_vcn:
+      dns_label: ocp
+
+    internet_gateway:
+      spec:
+        is_enabled: yes
+        display_name: "{{ cluster_state.infra_id }}-igw"
+
+    nat_gateway:
+      spec:
+        display_name: "{{ cluster_state.infra_id }}-natgw"
+      # route_table_name: "{{ cluster_state.infra_id }}-rt-private"
+          
+        # tags: "{{ cluster_state.tags | d({}) }}"
+        # wait: false
+    #   - name: "{{ cluster_state.infra_id }}-natgw-1b"
+    #     subnet: "{{ cluster_state.infra_id }}-net-public-1b"
+    #     tags: "{{ cluster_state.tags | d({}) }}"
+    #     wait: false
+    #   - name: "{{ cluster_state.infra_id }}-natgw-1c"
+    #     subnet: "{{ cluster_state.infra_id }}-net-public-1c"
+    #     tags: "{{ cluster_state.tags | d({}) }}"
+    #     wait: true
+
+    #> TODO use generic list
+    route_table_public:
+      spec:
+        display_name: "{{ cluster_state.infra_id }}-rt-public"
+      route_rules:
+      - spec:
+          destination: "0.0.0.0/0"
+          destination_type: "CIDR_BLOCK"
+        network_entity_type: internet-gateway
+        network_entity_name: "{{ cluster_state.infra_id }}-igw"
+
+    route_table_private:
+      spec:
+        display_name: "{{ cluster_state.infra_id }}-rt-private"
+      route_rules:
+      - spec:
+          destination: "0.0.0.0/0"
+          destination_type: "CIDR_BLOCK"
+        network_entity_type: nat-gateway
+        network_entity_name: "{{ cluster_state.infra_id }}-natgw"
+
+    # # Generic list
+    # route_tables:
+    #   - public: yes
+    #     spec:
+    #       display_name: "{{ cluster_state.infra_id }}-rt-public"
+    #     route_rules:
+    #     - spec:
+    #         destination: "0.0.0.0/0"
+    #         destination_type: "CIDR_BLOCK"
+    #       network_entity_type: internet-gateway
+    #       network_entity_name: "{{ cluster_state.infra_id }}-igw"
+
+    #   - public: no
+    #     spec:
+    #       display_name: "{{ cluster_state.infra_id }}-rt-private"
+    #     route_rules:
+    #     - spec:
+    #         destination: "0.0.0.0/0"
+    #         destination_type: "CIDR_BLOCK"
+    #       network_entity_type: nat-gateway
+    #       network_entity_name: "{{ cluster_state.infra_id }}-natgw"
+
+    # create permissive security Lists to force to use NSGs
+    # https://docs.oracle.com/en-us/iaas/tools/oci-ansible-collection/4.12.0/collections/oracle/oci/oci_network_security_list_module.html#ansible-collections-oracle-oci-oci-network-security-list-module
+    security_lists:
+    - spec:
+        display_name: "{{ cluster_state.infra_id }}-seclist-public"
+        ingress_security_rules: []
+        # - protocol: all
+        #   source: 0.0.0.0/0
+        #   is_stateless: false
+        #   source_type: CIDR_BLOCK
+        #   description: Allow all inbound
+        egress_security_rules:
```

Comment:
To be reviewed. Maybe it's interesting to delegate permissions to NSG and make sure everything will works correctly. NSG is best practice.